CRM/Sepa/Page/MarkGroupReceived.php

```diff
@@ -52,25 +53,35 @@ function run() {
     }
 
     // run the 'mark received' process
+    $this->closeGroup($group_id);
+
+    // go back to the dashboard
+    CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/sepa/dashboard', 'status=closed'));
```

Comment:
Why is closing the group necessary here, only closed groups can be marked received …?

Reply:
I've renamed the method to `markGroupAsReceived()`. (The code is the same as before, but with locking.)